react/features/base/devices/functions.js

```diff
@@ -13,6 +38,71 @@ export function getAudioOutputDeviceId() {
     return JitsiMeetJS.mediaDevices.getAudioOutputDevice();
 }
 
+/**
+ * Finds a device with a label that matches the passed label and returns its id.
+ *
+ * @param {Object} state - The redux state.
+ * @param {string} label - The label.
+ * @returns {string|undefined}
+ */
+export function getDeviceIdByLabel(state: Object, label: string) {
+    const types = [ 'audioInput', 'audioOutput', 'videoInput' ];
+
+    for (const type of types) {
+        const device
+            = state['features/base/devices'].devices[type]
+                .find(d => d.label === label);
+
+        if (device) {
+            return device.deviceId;
+        }
+    }
```

Comment:
Have you thought of changing lib-jitsi-meet to support taking in a label? There are other problems there I assume though, like how it's currently not using exact constraint matching.

Reply:
I haven't. This will be nice I guess.